examples/pytorch/ogb/directional_GSN/main.py

```diff
@@ -0,0 +1,433 @@
+    test_loader = GraphDataLoader(testset, batch_size=params.batch_size, shuffle=False, collate_fn=dataset.collate, pin_memory=True)
+
+    with tqdm(range(params.epochs), unit='epoch') as t:
+        for epoch in t:
+            t.set_description('Epoch %d' % epoch)
+
+            epoch_train_loss, epoch_train_ap, optimizer = train_epoch(model, optimizer, device, train_loader)
+            epoch_val_loss, epoch_val_ap = evaluate_network(model, device, val_loader)
+
+            epoch_train_losses.append(epoch_train_loss)
+            epoch_val_losses.append(epoch_val_loss)
+            epoch_train_APs.append(epoch_train_ap.item())
+            epoch_val_APs.append(epoch_val_ap.item())
+
+            _, epoch_test_ap = evaluate_network(model, device, test_loader)
+
+            epoch_test_APs.append(epoch_test_ap.item())
+
+            t.set_postfix(train_loss=epoch_train_loss, 
+                          train_AP=epoch_train_ap.item(), val_AP=epoch_val_ap.item(),
+                          refresh=False)
+
+            scheduler.step(-epoch_val_ap.item())
+
+            if optimizer.param_groups[0]['lr'] < params.min_lr:
+                print("\n!! LR EQUAL TO MIN LR SET.")
+                break
+
+            print('')
+
+    best_val_epoch = np.argmax(np.array(epoch_val_APs))
+    best_train_epoch = np.argmax(np.array(epoch_train_APs))
+    best_val_ap = epoch_val_APs[best_val_epoch]
+    best_val_test_ap = epoch_test_APs[best_val_epoch]
+    best_val_train_ap = epoch_train_APs[best_val_epoch]
+    best_train_ap = epoch_train_APs[best_train_epoch]
+
+    print("Best Train AP: {:.4f}".format(best_train_ap))
+    print("Best Val AP: {:.4f}".format(best_val_ap))
+    print("Test AP of Best Val: {:.4f}".format(best_val_test_ap))
+    print("Train AP of Best Val: {:.4f}".format(best_val_train_ap))
+
+class PCBADataset(Dataset):
+    def __init__(self, name):
+        print("[I] Loading dataset %s..." % (name))
+        self.name = name
+        
+        self.dataset, self.split_idx = prepare_dataset(name)
+        print("One hot encoding substructure counts... ", end='')
+        self.d_id = [1]*self.dataset[0][0].edata['subgraph_counts'].shape[1]
+
+        for g in self.dataset:
+            g[0].edata['eig'] = g[0].edata['subgraph_counts'].float()
+            
+        self.raw_train = Subset(self.dataset, self.split_idx['train'])
+        self.raw_val = Subset(self.dataset, self.split_idx['valid'])
+        self.raw_test = Subset(self.dataset, self.split_idx['test'])
+
+        self.train, self.val, self.test = [], [], []
+        for g in self.raw_train:
+            if g[0].num_nodes() > 5:
+                self.train.append(g)
+        for g in self.raw_val:
+            if g[0].num_nodes() > 5:
+                self.val.append(g)
+        for g in self.raw_test:
+            if g[0].num_nodes() > 5:
+                self.test.append(g)
```

Comment:
You can define a `Subset` class yourself that also handles this `num_nodes` check, so that you do not need to have both `raw_train` and `train`, etc.